server/config/passport.ts

```diff
@@ -27,10 +27,19 @@ passportUse(new LocalStrategy({
         .select('username email hashed_password salt')
         .exec((err, user) => {
             if (err) return done(err);
+            let errorObject = { message: 'wrong username or password' };
             if (!user || !user.authenticate(password)) {
-                return done(null, false, { message: 'wrong username or password' });
+                return done(null, false, errorObject);
+            } else {
+                user.authenticate(password)
+                    .then(res => {
+                        if (res) {
+                            done(null, user);
+                        } else {
+                            done(null, false, errorObject);
+                        }
+                    });
```

Comment:
I am not sure to understand this change.

If `!user` is true, then you return your error, which seems legit. If `!user` is false, then you try to authenticate. In case it fails, you send your error… but otherwise, you try to authenticate again?

Probably what you really wanted to write was :

```
            let errorObject = { message: 'wrong username or password' };
            if (!user) {
                return done(null, false, errorObject);
            } else {
                user.authenticate(password)
                    .then(res => {
                        if (res) {
                            done(null, user);
                        } else {
                            done(null, false, errorObject);
                        }
                    });
```

No?